authors/apps/notifications/backends.py

```diff
@@ -0,0 +1,199 @@
+                    person.profile.notification_perms:
+                people_to_notify.append(person.profile)
+        for person in people_to_notify:
+            action_link = request.build_absolute_uri(
+                reverse('articles:article-update',
+                        kwargs={'slug': article.slug}))
+
+            message = self._message_builder(article.author, 
+                                            "published a new article",
+                                            article.title)
+            self._dispatch(request,
+                           (article.author, person, message, action_link,),
+                           'FOLLOWED_AUTHOR_ARTICLE_CREATED')
+
+    def article_interaction_comment(self, request, comment):
+        """
+        notify users that created or favorited an article of
+        a new comment
+
+        args:
+            - request (Request): the commenter's request object
+            - comment (CommentModel): The comment object
+
+        """
+        sender = comment.author
+        receiver = comment.article.author
+        if self._is_same_profile(sender, receiver):
+            return
+        message = self._message_builder(sender,
+                                        "left a  comment on your article",
+                                        comment.article.title
+                                        )
+        action_link = request.build_absolute_uri(reverse(
+            'comments:comment-detail', 
+            kwargs={'slug': comment.article.slug, 'pk': comment.pk}))
+        self._dispatch(request, (sender,
+                       receiver, message, action_link,),
+                       'AUTHORED_ARTICLE_COMMENTED')
+
+        for profile in comment.article.favorites.all():
+            if 'FAVORITED_ARTICLE_COMMENTED' in profile.notification_perms:
+                message = self._message_builder(
+                    sender, "left a  comment on an article you favorited,",
+                    comment.article.title
+                )
+                profile.user.email
+                self._dispatch(request, (sender,
+                               profile, message, action_link,),
+                               'FAVORITED_ARTICLE_COMMENTED')
+
+    def article_interaction_liked_or_faved(
+            self, request, article, liked_by=None, favorited_by=None):
+        """
+        notify users that created an article of
+        a new like or favorite
+
+        args:
+            - request (Request): the commenter's request object
+            - article(ArticleModel): the article object
+            - liked_by (ProfileModel): the liker's profile object, Defaults to None
+            - favorited_by (ProfileModel): the favoriter's profile object, Defaults to None  
+
+        """
+        sender = liked_by or favorited_by
+        receiver = article.author
+
+        if self._is_same_profile(sender, receiver):
+            return
```

Comment:
@malfahad Is there a specific reason for this `falsey return`? I think raising an exception would be better.

Reply:
Yes, for example, comments, The notification creation is dispatched when a comment is saved. Take for example creating a comment on your own article. This falsey return stops the app from notifying one about their own actions.